plugins/logitech-rallysystem/fu-logitech-rallysystem-audio-device.c

```diff
@@ -0,0 +1,279 @@
+/*
+ * Copyright (c) 1999-2023 Logitech, Inc.
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "config.h"
+
+#include <linux/types.h>
+#include <linux/usb/video.h>
+#include <linux/uvcvideo.h>
+#ifdef HAVE_IOCTL_H
+#include <linux/hidraw.h>
+#include <sys/ioctl.h>
+#endif
+
+#include <string.h>
+
+#include "fu-logitech-rallysystem-audio-device.h"
+
+#define FU_LOGITECH_RALLYSYSTEM_AUDIO_DEVICE_IOCTL_TIMEOUT 50000 /* ms */
+#define TOPOLOGY_DATA_LEN				   513 /* 512 plus 1 byte for the hid report id */
+#define SERIAL_NUMBER_REQUEST_DATA_LEN			   49
+#define SERIAL_NUMBER_RESPONSE_DATA_LEN			   128
+#define SERIAL_NUMBER_DATA_LEN				   12
+
+#ifndef HIDIOCGINPUT
+#define HIDIOCGINPUT(len) _IOC(_IOC_READ, 'H', 0x0A, len)
+#endif
```

Comment:
what OS is this needed for?  I would think the plugin just shouldn't compile if this is missing.